src/App.js

```diff
@@ -217,6 +225,10 @@ class App extends Component {
     }
   };
 
+  handleWelcomePageClose = () => {
+    this.setState({ welcomeShown: true });
```

Comment:
here seems a good place...

Reply:
yes... exactly...  the welcome page is "shown" when the user voluntarily closes the dialog or change page. That means that the user has seen it. Otherwise the app may had crashed and the welcome page was not seen yet